apps/server/src/modules/task/uc/task.authorization.service.ts

```diff
@@ -9,22 +9,40 @@ export enum TaskParentPermission {
 
 @Injectable()
 export class TaskAuthorizationService {
-	constructor(private readonly courseRepo: CourseRepo) {}
-
-	/**
-	 * Important user group operations are only a temporary solution until we have established groups
-	 */
-	async getPermittedCourses(userId: EntityId, permission: TaskParentPermission): Promise<EntityId[]> {
-		// courseGroups are missing
-		// lessons are missing -> only search for hidden: false,
+	constructor(private readonly courseRepo: CourseRepo, private readonly lessonRepo: LessonRepo) {}
+
+	// it should return also the scopePermissions for this user added to the entity .scopePermission: { userId, read: boolean, write: boolean }
+	// then we can pass and allow only scoped courses to getPermittedLessonIds and validate read write of .scopePermission
+	async getPermittedCourses(userId: EntityId, neededPermission: TaskParentPermission): Promise<Course[]> {
 		let permittedCourses: Course[] = [];
-		if (permission === TaskParentPermission.write) {
+
+		if (neededPermission === TaskParentPermission.write) {
 			[permittedCourses] = await this.courseRepo.findAllForTeacher(userId);
-		} else if (permission === TaskParentPermission.read) {
-			[permittedCourses] = await this.courseRepo.findAllForStudent(userId);
+		} else if (neededPermission === TaskParentPermission.read) {
+			[permittedCourses] = await this.courseRepo.findAllByUserId(userId);
 		}
 
-		const entityIds = permittedCourses.map((o) => o.id);
-		return entityIds;
+		return permittedCourses;
+	}
+
+	async getPermittedLessons(userId: EntityId, courses: Course[]): Promise<Lesson[]> {
+		const writeCourses = courses.filter(
+			(c) => c.getSubstitutionTeacherIds().includes(userId) || c.getTeacherIds().includes(userId)
+		);
+		const readCourses = courses.filter((c) => c.getStudentIds().includes(userId));
+
+		const writeCourseIds = writeCourses.map((c) => c.id);
+		const readCourseIds = readCourses.map((c) => c.id);
+
+		// idea as combined query:
+		// [{courseIds: onlyWriteCoursesIds}, { courseIds: onlyReadCourses, filter: { hidden: false }}]
+		const [[writeLessons], [readLessons]] = await Promise.all([
```

Comment:
We have no real "permission" for it. We have only a parent and they say which the user can and should see. It is implemented wrong in the task/ route. Because it persist it to permission and roles and must change in future. We rename homework in task that for example the student in learnroom "Arbeitsgruppe Fussball" can create a task an work with it. 

In client is is hard persisted and can not changed at the moment. But we should not do the same for the server.